JSDOMParser.js

```diff
@@ -1022,10 +1022,17 @@
             continue;
           }
           if (nextC === "/" && this.html.substr(this.currentChar, 8 /*"/script>".length */).toLowerCase() == "/script>") {
-            // Go back before the '<' so we find the end tag.
-            this.currentChar--;
-            // Done with this script tag, the caller will close:
-            return;
+            var indexOfNextScriptOpeningTag = this.html.indexOf("<script", this.currentChar);
+            var indexOfNextScriptClosingTag = this.html.indexOf("</script>", this.currentChar);
+            // Found out the closing script tag when there is no other opening/closing tag or
+            // index of closing tag is bigger than opening tag's, which means rest of opening/closing script tags are pairs.
```

Comment:
This isn't actually true... What happens to the first opening and closing tag means nothing for whether *all* the next opening/closing tags are pairs.